sdks/go/pkg/beam/transforms/top/top.go

```diff
@@ -157,10 +154,13 @@ func accumEnc() func(accum) ([]byte, error) {
 		panic(err)
 	}
 	return func(a accum) ([]byte, error) {
-		if a.enc == nil {
-			return nil, errors.Errorf("top.accum: element encoder unspecified")
+		if len(a.list) > 0 && a.enc == nil {
+			return nil, errors.Errorf("top.accum: element encoder unspecified with non-zero elements: %v data available", len(a.data))
 		}
 		var values [][]byte
+		if len(a.list) == 0 && len(a.data) > 0 {
+			values = a.data
+		}
```

Comment:
More of an understanding question if `a.data` could also be assigned to nil as we do for `a.list` after this if block

Reply:
That's a very good question.

My instinct is that since this function is supposed to write and encode data, it shouldn't be mutating the value at all. 

Which means technically it shouldn't be nil'ing out that list field at all either. 

In principle, the encoding will only happen when emitting the values downstream, which should only happen once per given value, at which point the value itself should be garbage collected away anyway.

So I'm going to do the opposite: remove the nil on encode line there.